lib/std/mem.zig

```diff
@@ -1013,29 +1006,25 @@ pub fn indexOfScalar(comptime T: type, slice: []const T, value: T) ?usize {
 /// Linear search for the last index of a scalar value inside a slice.
 pub fn lastIndexOfScalar(comptime T: type, slice: []const T, value: T) ?usize {
     var i: usize = slice.len;
-    while (i != 0) {
+    return while (i != 0) {
         i -= 1;
-        if (slice[i] == value) return i;
-    }
-    return null;
+        if (slice[i] == value) break i;
+    } else null;
 }
 
 pub fn indexOfScalarPos(comptime T: type, slice: []const T, start_index: usize, value: T) ?usize {
-    var i: usize = start_index;
-    while (i < slice.len) : (i += 1) {
-        if (slice[i] == value) return i;
-    }
-    return null;
+    return for (start_index..slice.len) |i| {
+        if (slice[i] == value) break i;
+    } else null;
 }
 
 pub fn indexOfAny(comptime T: type, slice: []const T, values: []const T) ?usize {
     return indexOfAnyPos(T, slice, 0, values);
 }
 
 pub fn lastIndexOfAny(comptime T: type, slice: []const T, values: []const T) ?usize {
-    var i: usize = slice.len;
-    while (i != 0) {
-        i -= 1;
+    var i: usize = slice.len - 1;
+    while (i != 0) : (i -= 1) {
```

Comment:
This change is just incorrect: this loop has different semantics. It will run for one iteration less, skipping 0.

Reply:
Reverted and fixed now.